subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java

```diff
@@ -52,6 +52,8 @@
 
     void setImplementationClasspath(List<URL> implementationClasspath);
 
+    void enableProcessInfoPublishing(boolean shouldPublish);
```

Comment:
Maybe rename to indicate the JVM memory will be collected, and `WorkerProcess.getJvmMemoryStatus()` should probably fail if this was not enabled.